packages/PieChart/src/PieChart.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import cx from 'classnames';
+import { PieChart as RPieChart, Cell, Label, Pie } from 'recharts';
+
+import tokens from '@igloo-ui/tokens/dist/base10/tokens.json';
+
+import './pie-chart.scss';
+
+export interface DataProps {
+  /** The class name of the data element in the chart */
+  className?: string;
+  /** The color of the data element */
+  color?: string;
+  /** The ID of the data element */
+  id: string;
+  /** The percentage value of the data element */
+  percentage: number;
+}
+
+export interface PieChartProps extends React.ComponentProps<'div'> {
+  /** The class name of the component */
+  className?: string;
+  /** The data to be displayed in the chart */
+  data?: DataProps[];
+  /** Add a data-test tag for automated tests */
+  dataTest?: string;
+  /** The label displayed in the center of the chart
+   * (This is not displayed if the chart size is "regular")
+   */
+  label?: React.ReactNode;
+  /** The rate of the chart in percentage
+   * (This is not displayed if the chart size is "regular")
+   */
+  rate?: number;
+  /** The size of the chart */
+  size?: 'regular' | 'large';
+}
+
+const CustomLabel = (labelObj: any): React.ReactElement => {
+  const { viewBox, rate, label } = labelObj;
+  const { cx, cy } = viewBox;
+  const halfOfSymbolFontSize = 9;
+  const halfOfRateHeight = '2.8rem';
+
+  return (
+    <text
+      x={cx}
+      y={cy - 5}
+      fill="#3d405c"
+      className="recharts-text recharts-label"
+      textAnchor="middle"
+      dominantBaseline="central"
+      alignmentBaseline="middle"
+    >
+      <tspan
+        x={cx + halfOfSymbolFontSize}
+        alignmentBaseline="middle"
+        className="ids-pie-chart__rate"
+      >
+        {rate}
+      </tspan>
+      <tspan
+        baselineShift="super"
+        alignmentBaseline="middle"
+        className="ids-pie-chart__symbol"
+      >
+        %
+      </tspan>
+      <tspan x={cx} dy={halfOfRateHeight} className="ids-pie-chart__label">
+        {label}
```

Comment:
Do we want to limit the size of the label on support ellipsis for situations like this:
![image](https://github.com/gsoft-inc/ov-igloo-ui/assets/4323800/70f92ae9-dff2-4ad4-a186-e14b1dd01bf9)

Reply:
Good point. For percentage, we could probably limit the number between 0-100.
Should the label actually have a character limit, or just put an ellipsis using css? I vote for css, since the label can be anything, not just a string. What do you think?

Reply:
I vote for CSS too.

Reply:
well I couldn't find a way to actually type the percentage so the user will just need to type it in properly. They shouldn't go above 100